frontend/src/components/Dashboard/ScoresTableSection/index.jsx

```diff
@@ -83,18 +91,22 @@ class ScoresTable extends React.Component {
         </TableHeader>
         <TableBody displayRowCheckbox={false}>
         {
-          studentsList.map((studentInstance, studentIndex) => (
+          // Going over students
+          Object.keys(rebuiltStudentsList).map((studentId, studentIndex) => (
             <TableRow key={studentIndex}>
-              <TableRowColumn>{studentInstance.student.name}</TableRowColumn>
+              <TableRowColumn>{rebuiltStudentsList[studentId].studentName}</TableRowColumn>
               {
-                Object.keys(neededSubjects).map((subjectId, subjectIndex) => (
+                // Going over subjects
+                Object.keys(groupSubjectsList).map((subjectId, subjectIndex) => (
                   <TableRowColumn key={subjectIndex}>
                     <TextField
-                      id={studentInstance._id + '/' + subjectId}
+                      id={subjectId + '/' + studentId}
```

Comment:
Ooups. Just found out that providing the id for TextField was vital for material-ui in order to actually render it. So, you  now can merge the branches.